econml/cate_estimator.py

```diff
@@ -392,8 +392,11 @@ def marginal_effect_inference(self, T, X=None):
         if X is None:
             pred = np.repeat(pred, shape(T)[0], axis=0)
             pred_stderr = np.repeat(pred_stderr, shape(T)[0], axis=0)
-        return InferenceResults(d_t=cme_inf.d_t, d_y=cme_inf.d_y, pred=pred,
-                                pred_stderr=pred_stderr, inf_type='effect', pred_dist=None, fname_transformer=None)
+        # TODO: It seems wrong to return inference results based on a normal approximation
+        #       even in the case where const_marginal_effect_inference was actually generated
+        #       using bootstrap
+        return NormalInferenceResults(d_t=cme_inf.d_t, d_y=cme_inf.d_y, pred=pred,
```

Comment:
This should be checking what the inference type is and return the same type as what cme_inf is. So if it is EmpiricalInference, return EmpiricalInference.

Reply:
Fixed in the latest commit; now `const_marginal_effect_interval` and `const_marginal_effect_inference` both take an additional optional named argument `n_row` and they are responsible for sizing the output accordingly; `marginal_effect_interval` (and inference) then just defer to that implementation and don't need to worry about any resizing themselves.